doc/example/getting_started.cpp

```diff
@@ -112,18 +167,49 @@ MOCK_BASE_CLASS( mock_view, view )
 
 class calculator
 {
+    view& v;
 public:
-    calculator( view& v );
+    calculator( view& v ): v(v) {}
+
+    void add( int a, int b ){ v.display(a + b); }
+};
+
+struct CatchFailureFixture
+{
+    static bool aborted;
+    static std::string fail_msg;
 
-    void add( int a, int b );
+    static void abort()
+    {
+        aborted = true;
+    }
+    static void fail( const std::string& message, const std::string&, const char* = "", int = 0 ){
+        fail_msg = message;
+    }
+    CatchFailureFixture()
+    {
+        error_handler_abort = abort;
+        error_handler_fail = fail;
+    }
+    void assert_failure(const std::string& required_message)
+    {
+        BOOST_CHECK(aborted);
+        BOOST_CHECK(fail_msg.find(required_message) != std::string::npos);
+    }
 };
+bool CatchFailureFixture::aborted = false;
+std::string CatchFailureFixture::fail_msg;
+
 //[ action_test
-BOOST_AUTO_TEST_CASE( zero_plus_zero_is_zero )
+BOOST_FIXTURE_TEST_CASE( zero_plus_zero_is_zero_with_action, CatchFailureFixture )
 {
    mock_view v;
    calculator c( v );
    MOCK_EXPECT( v.display ).once().with( 0 ); // missing returns( true )
    c.add( 0, 0 );
+   assert_failure("missing action");
```

Comment:
This is showing up in the docs, I don’t think this was intentional, was it?

Reply:
It is. The docs say that at this point a test assertion fails that an action is missing. This highlights it AND actually tests it

Reply:
I’m thinking some newcomers will be puzzled by this line, especially since the definition for `assert_failure` will not be visible.
There should maybe at least be a comment?

Reply:
If we upgrade the required boost version to 1.59 (or higher) we can use assertions in fixtures teardown

- Initial support via `BOOST_AUTO_TEST_CASE(test, *utf::fixture<MixFixture>)` is in 1.59
- Better in 1.65 via  `BOOST_FIXTURE_TEST_CASE(test, MyFixture)` (like now but teardown() is supported in 1.65+)

What shall it be?

Reply:
I think 1.65 is fine if that helps.

Reply:
Upgraded that for new CI